src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1484,25 +1485,21 @@ public static bool IsAtEndOfPattern(this SyntaxTree syntaxTree, SyntaxToken left
 
             static bool IsAtEndOfSwitchStatementPattern(SyntaxToken leftToken)
             {
-                var node = leftToken.Parent;
+                SyntaxNode node = leftToken.Parent as ExpressionSyntax;
+                if (node == null)
```

Comment:
```suggestion
                if (leftToken.Parent is not ExpressionSyntax node)
```

Reply:
This changes the type to be ExpressionSyntax. But it's needed to be SyntaxNode as we walk out of the expression.

Reply:
Ah correct. I should have suggested var to fix the nullability warning.

Reply:
Yup!  Not sure why I didn't get a warning in the ide